test/test_jit.py

```diff
@@ -6271,6 +6271,19 @@ def forward(self, x):
             mte, (torch.zeros(1, 2, 3),), None, verbose=False,
             example_outputs=outputs))
 
+    def test_trace_nested_datatypes(self):
+        @torch.jit.script
+        def foo(x):
+            return [[x + 1, x - 1], [x + 2, x - 2]]
+
+        def bar(x):
+            list_stuff = foo(x)
+            return list_stuff[0][0], list_stuff[1][1]
+
+        traced = torch.jit.trace(bar, torch.rand(3, 4))
+        x = torch.rand(3, 4)
+        self.assertEqual(bar(x), traced(x))
```

Comment:
Can we check them on different inputs? This test would pass if we would have embedded all values as constants in the body of `bar`

Reply:
Not quite sure what you mean. We trace with one call of `rand()` and test with another. Could you clarify?